sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SimpleSyncPoller.java

```diff
@@ -0,0 +1,284 @@
+     *     if service supports cancellation, this parameter is required and if service does not support cancellation
+     *     then the implementer should return Mono.error with an error message indicating absence of cancellation
+     *     support, the operation will be called with current {@link PollingContext}.
+     * @param fetchResultOperation a {@link Function} that represents the  operation to retrieve final result of
+     *     the long running operation if service support it, this parameter is required and operation will be called
+     *     current {@link PollingContext}, if service does not have an api to fetch final result and if final result
+     *     is same as final poll response value then implementer can choose to simply return value from provided
+     *     final poll response.
+     */
+    public SimpleSyncPoller(Duration pollInterval,
+                     Function<PollingContext<T>, PollResponse<T>> syncActivationOperation,
+                     Function<PollingContext<T>, PollResponse<T>> pollOperation,
+                     BiFunction<PollingContext<T>, PollResponse<T>, T> cancelOperation,
+                     Function<PollingContext<T>, U> fetchResultOperation) {
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        if (pollInterval.isNegative() || pollInterval.isZero()) {
+            throw LOGGER.logExceptionAsWarning(new IllegalArgumentException(
+                "Negative or zero value for 'defaultPollInterval' is not allowed."));
+        }
+        this.pollInterval = pollInterval;
+        Objects.requireNonNull(syncActivationOperation, "'syncActivationOperation' cannot be null.");
+        this.pollOperation = Objects.requireNonNull(pollOperation, "'pollOperation' cannot be null.");
+        this.cancelOperation = Objects.requireNonNull(cancelOperation, "'cancelOperation' cannot be null.");
+        this.fetchResultOperation = Objects.requireNonNull(fetchResultOperation,
+            "'fetchResultOperation' cannot be null.");
+        this.activationResponse = syncActivationOperation.apply(this.pollingContext);
+        //
+        this.pollingContext.setOnetimeActivationResponse(this.activationResponse);
+        this.pollingContext.setLatestResponse(this.activationResponse);
+        if (this.activationResponse.getStatus().isComplete()) {
+            this.terminalPollContext = this.pollingContext;
+        }
+    }
+
+    @Override
+    public synchronized PollResponse<T> poll() {
+        PollResponse<T> response = this.pollOperation
+            .apply(this.pollingContext);
+        this.pollingContext.setLatestResponse(response);
+        if (response.getStatus().isComplete()) {
+            this.terminalPollContext = this.pollingContext.copy();
+        }
+        return response;
+    }
+
+    @Override
+    public PollResponse<T> waitForCompletion() {
```

Comment:
should this method call into waitForCompletion(null) and handle special case for null there? there's quite a duplication between this and overload that could be reduced.